File: Android/FoodRescue/app/src/main/java/edu/ewu/team1/foodrescue/firebase/MyFirebaseInstanceIDService.java
package edu.ewu.team1.foodrescue.firebase;

import android.util.Log;

import com.google.firebase.iid.FirebaseInstanceId;
import com.google.firebase.iid.FirebaseInstanceIdService;
import com.google.firebase.messaging.FirebaseMessaging;

import static android.content.ContentValues.TAG;

public class MyFirebaseInstanceIDService extends FirebaseInstanceIdService {
    public MyFirebaseInstanceIDService() {
    }

    /**
     * Refresh this devices firebase token and subscribe to the topic
     */
    @Override
    public void onTokenRefresh() {
        // Get updated InstanceID token.
        String refreshedToken = FirebaseInstanceId.getInstance().getToken();
        Log.d(TAG, "Token: " + refreshedToken);
        FirebaseMessaging.getInstance().subscribeToTopic("edu.ewu.team1.foodrescue");
    }
}

File: Android/FoodRescue/app/src/main/java/edu/ewu/team1/foodrescue/firebase/MyFirebaseInstanceIDService.kt
package edu.ewu.team1.foodrescue.firebase

import android.util.Log

import com.google.firebase.iid.FirebaseInstanceId
import com.google.firebase.iid.FirebaseInstanceIdService
import com.google.firebase.messaging.FirebaseMessaging

import android.content.ContentValues.TAG

class MyFirebaseInstanceIDService : FirebaseInstanceIdService() {

    /**
     * Refresh this devices firebase token and subscribe to the topic
     */
    override fun onTokenRefresh() {
        // Get updated InstanceID token.
        val refreshedToken = FirebaseInstanceId.getInstance().token
        Log.d(TAG, "Token: " + refreshedToken!!)
        FirebaseMessaging.getInstance().subscribeToTopic("edu.ewu.team1.foodrescue")
    }
}
